src/commands/kol/player.ts

```diff
@@ -0,0 +1,84 @@
+import {
+    ChatInputCommandInteraction,
+    SlashCommandBuilder,
+    bold, 
+    hyperlink
+  } from "discord.js";
+  
+import { kolClient } from "../../clients/kol";
+import { pluralize, toKoldbLink, toSnapshotLink } from "../../utils";
+
+
+export const data = new SlashCommandBuilder()
+  .setName("player")
+  .setDescription("Look up information on a given player.")
+  .addStringOption((o) =>
+    o
+      .setName("player")
+      .setDescription("The KOL username of the player you're looking up.")
+      .setRequired(true)
+    );
+
+export async function execute(interaction: ChatInputCommandInteraction) {
+    const member = interaction.member;
+  
+    if (!member) {
+      interaction.reply({
+        content: "You have to perform this action from within a Guild.",
+        ephemeral: true,
+      });
+      return;
+    }
+
+    const playerNameOrId = interaction.options.getString("player", true);
+
+    await interaction.deferReply();
+
+    const player = await kolClient.getPartialPlayer(playerNameOrId);
+
+    if (!player) {
+        interaction.editReply({content:"According to KOL, this player does not exist."});
+        return;
+    }
+
+    const playerInfo = await kolClient.getPlayerInformation(player.id);
+
+    if (!playerInfo) {
+        interaction.editReply({content:"While this player exists, this command didn't work. Weird."});
+        return;
+    }
+
+    const playerInfoFormatted = [
+        `This user is currently a ${bold(player.class)} at level ${player.level}`,
+        `They have completed ${bold(
+            hyperlink(
+                    `${playerInfo.ascensions}`,
+                    toKoldbLink(player.name)))} 
+            ${pluralize(playerInfo.ascensions,"ascension","ascensions")}.`,
+        `Loves eating ${playerInfo.favoriteFood}.`,
+        `Loves drinking ${playerInfo.favoriteBooze}.`,
+        ``,
+        `If it exists, their av-snapshot is located ${hyperlink(
+            "here", toSnapshotLink(player.name))}`,
+        ``,
```

Comment:
I'm assuming these empty strings are meant for spacing, but there is probably a more elegant way to address that.

Reply:
there definitely is, format-wise this is pretty shit but i just wanted to put a starting point out there and see if sam or you had better ideas there. i could just do an output.push() type thing like sam does in getItemDescription, that'd be a little more elegant probably.

Reply:
resolving because the embedbuilder version is better although i'm still skeptical it's good enough